RockWeb/App_Code/GetEventCalendarFeed.cs

```diff
@@ -115,70 +117,87 @@ private iCalendar CreateICalendar( CalendarProps calendarProps )
-                            string contactEmail = !string.IsNullOrEmpty( occurrence.ContactEmail ) ? ", Email: " + occurrence.ContactEmail : string.Empty;
-                            string contactPhone = !string.IsNullOrEmpty( occurrence.ContactPhone ) ? ", Phone: " + occurrence.ContactPhone : string.Empty;
-                            string contactInfo = contactName + contactEmail + contactPhone;
-
-                            ievent.Contacts.Add( contactInfo );
-                            ievent.Comments.Add( contactInfo );
-                        }
+                        iCalendarSerializer serializer = new iCalendarSerializer();
+                        iCalendarCollection ical = (iCalendarCollection)serializer.Deserialize( occurrence.Schedule.iCalendarContent.ToStreamReader() );
 
-                        // TODO: categories - comma delimited list of whatever, might use audience
-                        foreach ( var a in eventItem.EventItemAudiences )
+                        foreach ( var icalEvent in ical[0].Events )
                         {
-                            ievent.Categories.Add( a.DefinedValue.Value );
+                            // We get all of the schedule info from Schedule.iCalendarContent
+                            Event ievent = icalEvent.Copy<Event>();
+
+                            ievent.Summary = !string.IsNullOrEmpty( eventItem.Name ) ? eventItem.Name : string.Empty;
+                            ievent.Location = !string.IsNullOrEmpty( occurrence.Location ) ? occurrence.Location : string.Empty;
+
+                            ievent.DTStart.SetTimeZone( icalendar.TimeZones[0] );
+                            ievent.DTEnd.SetTimeZone( icalendar.TimeZones[0] );
+
+                            // Rock has more descriptions than iCal so lets concatenate them
+                            string description = CreateEventDescription( eventItem, occurrence );
+
+                            // Don't set the description prop for outlook to force it to use the X-ALT-DESC property which can have markup.
+                            if ( interactionDeviceType != "Outlook" )
+                            {
+                                ievent.Description = description.ConvertBrToCrLf()
+                                                                    .Replace( "</P>", "" )
+                                                                    .Replace( "</p>", "" )
+                                                                    .Replace( "<P>", Environment.NewLine )
+                                                                    .Replace( "<p>", Environment.NewLine )
+                                                                    .Replace( "&nbsp;", " " )
+                                                                    .SanitizeHtml();
+                            }
```

Comment:
The description formatting changed to remove HTML code that was showing up when viewing the feed in a Google calendar.